src/ui/components.tsx

```diff
@@ -30,7 +30,7 @@ interface CheckMkAsyncSelectProps<T> {
   width?: number;
   autocompleter: (prefix: string) => Promise<Array<SelectableValue<NonNullable<T>>>>;
   onChange: (value: T) => void;
-  value: T;
+  value?: T;
```

Comment:
sorry, that i find something new on each iteration...

this seems to be a bit broad to me. what do you think of the following change:

```diff
diff --git a/src/ui/components.tsx b/src/ui/components.tsx
index 1615d15..4a42954 100644
--- a/src/ui/components.tsx
+++ b/src/ui/components.tsx
@@ -30,7 +30,7 @@ interface CheckMkAsyncSelectProps<T> {
   width?: number;
   autocompleter: (prefix: string) => Promise<Array<SelectableValue<NonNullable<T>>>>;
   onChange: (value: T) => void;
-  value?: T;
+  value: T;
   inputId: string; // make the InlineField magic do its work // TODO: find better solution
 }
 
@@ -155,7 +155,7 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
         }
       : { ...props.value };
 
-  const onValueChange = (newValue: string) => {
+  const onValueChange = (newValue: string | undefined) => {
     if (newValue === undefined) {
       throw 'this is an internal error, please report';
     }
@@ -171,7 +171,7 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
   return (
     <HorizontalGroup>
       <InlineField label={label} labelWidth={14}>
-        <CheckMkAsyncSelect
+        <CheckMkAsyncSelect<string | undefined>
           inputId={`input_${props.label}`}
           label={label}
           autocompleter={autocompleter}
```

this would be a more specific to the thing we want to change?!

Reply:
I like it! A lot!